server/routing/pubsub/messages.go

```diff
@@ -0,0 +1,110 @@
+// Copyright AGNTCY Contributors (https://github.com/agntcy)
+// SPDX-License-Identifier: Apache-2.0
+
+package pubsub
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+)
+
+// LabelAnnouncement is the wire format for label announcements via GossipSub.
+// This is a minimal structure optimized for network efficiency.
+//
+// Protocol parameters: See constants.go for TopicLabels, MaxMessageSize, etc.
+// These are intentionally NOT configurable to ensure network-wide compatibility.
+//
+// Conversion to storage format:
+//   - Wire: LabelAnnouncement with []string labels
+//   - Storage: Enhanced keys (/skills/AI/CID/PeerID) with labels.LabelMetadata
+//
+// Example:
+//
+//	{
+//	  "cid": "bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi",
+//	  "peer_id": "12D3KooWD3bfmNbuuuT5Zch8fj9Cg9dQR2FpGm7JzCfCzPWZnxLn",
+//	  "labels": ["/skills/AI/ML", "/domains/research", "/modules/tensorflow"],
+//	  "timestamp": "2025-10-01T10:00:00Z"
+//	}
+type LabelAnnouncement struct {
```

Comment:
i would suggest using explicit event for record publications, eg


```suggestion
type RecordAnnouncement struct {
```

or 


```suggestion
type RecordPublishEvent struct {
```

Reply:
fixed